lib/events.js

```diff
@@ -735,37 +767,38 @@ EventEmitter.prototype.off = EventEmitter.prototype.removeListener;
  */
 EventEmitter.prototype.removeAllListeners =
     function removeAllListeners(type) {
-      const events = this._events;
+      const events = this[kEvents];
       if (events === undefined)
         return this;
 
       // Not listening for removeListener, no need to emit
-      if (events.removeListener === undefined) {
+      if (!events.has('removeListener')) {
         if (arguments.length === 0) {
-          this._events = { __proto__: null };
-          this._eventsCount = 0;
-        } else if (events[type] !== undefined) {
-          if (--this._eventsCount === 0)
-            this._events = { __proto__: null };
+          // this[kEvents].clear();
+          this[kEvents]= new Map();
+        } else if (events.has(type)) {
+          if (this[kEvents].size === 1)
+            // this[kEvents].clear();
+            this[kEvents] = new Map();
           else
-            delete events[type];
+            this[kEvents].delete(type);
         }
         return this;
       }
 
       // Emit removeListener for all listeners on all events
       if (arguments.length === 0) {
-        for (const key of ReflectOwnKeys(events)) {
+        for (const key of events.keys()) {
           if (key === 'removeListener') continue;
           this.removeAllListeners(key);
         }
         this.removeAllListeners('removeListener');
-        this._events = { __proto__: null };
-        this._eventsCount = 0;
+        // this[kEvents].clear();
+        this[kEvents] = new Map();
```

Comment:
```suggestion
        this[kEvents] = new SafeMap();
```

Reply:
Thanks, I planned to change it after all the benchmarking